rules/legacy/src/Rector/Include_/AddTopIncludeRector.php

```diff
@@ -27,74 +24,34 @@
 final class AddTopIncludeRector extends AbstractFileSystemRector
 {
     /**
-     * @var string
+     * @var String_
      */
-    private $file = '';
+    private $autoloadFilePathString;
 
     /**
-     * @var ?Expr
-     */
-    private $fileExpression;
-
-    /**
-     * @var int
-     */
-    private $type = 0;
-
-    /**
-     * @var string[] to match
+     * @var string[]
      */
     private $patterns = [];
 
-    /**
-     * @var Parser
-     */
-    private $parser;
-
-    /**
-     * @var Standard
-     */
-    private $standard;
-
-    public function __construct(string $type = 'TYPE_INCLUDE', string $file = '"autoload.php"', array $match = [])
+    public function __construct(string $autoloadFilePath = '/autoload.php', array $match = [])
     {
-        $this->parser = (new ParserFactory())->create(ParserFactory::PREFER_PHP7);
-        $this->standard = new Standard();
-        $reflection = new ReflectionClass(Include_::class);
-        $constants = $reflection->getConstants();
-
-        if (! isset($constants[$type])) {
-            throw new InvalidRectorConfigurationException('Invalid type: must be one of ' . implode(
-                ', ',
-                array_keys($constants)
-            ));
-        }
-
-        $this->type = $constants[$type];
-        $this->file = $file;
-        $this->fileExpression = $this->getExpressionFromString($this->file);
-        // normalize the file we are including
-        $this->file = $this->getStringFromExpression($this->fileExpression);
+        $this->autoloadFilePathString = new String_($autoloadFilePath);
         $this->patterns = $match;
     }
 
     public function getDefinition(): RectorDefinition
     {
-        return new RectorDefinition('Adds an include file at the top of matching files but not class definitions', [
+        return new RectorDefinition('Adds an include file at the top of matching files, except class definitions', [
             new CodeSample(
                 <<<'PHP'
-<?php
-
-if (isset($_POST["csrf"])) {
+if (isset($_POST['csrf'])) {
     processPost($_POST);
 }
 PHP,
                 <<<'PHP'
-<?php
-
-include "autoloader.php";
+require_once __DIR__ . '/../autoloader.php';
```

Comment:
Looking at include/require, the `require_once` makes the only sense.

- `include` allows missing the file, that would be false positive
- and `require` (without `_once`) can include the file twice, so it would break it